packages/cx-layout/src/components/Header/HeaderClean.tsx

```diff
@@ -12,94 +12,45 @@
  * limitations under the License.
  */
 
-import React, {
-  FC, HTMLProps,
-} from 'react';
-import {
-  DesignableComponentsProps,
-  Div,
-  Fragment,
-  designable,
-  ComponentOrTag,
-} from '@bodiless/fclasses';
+import React, { FC } from 'react';
+import { Div, Fragment, designable } from '@bodiless/fclasses';
 import { asCxTokenSpec } from '@bodiless/cx-elements';
 import { LogoClean } from '../Logo';
-
-export type HeaderComponents = {
-  Wrapper: ComponentOrTag<any>,
-  Container: ComponentOrTag<any>,
-  MenuContainer: ComponentOrTag<any>,
-  MenuToggler: ComponentOrTag<any>,
-  Menu: ComponentOrTag<any>,
-  Logo: ComponentOrTag<any>,
-  UtilityMenu: ComponentOrTag<any>,
-  SocialLinks: ComponentOrTag<any>,
-  ButtonsMenuWrapper: ComponentOrTag<any>,
-  ButtonsWrapper: ComponentOrTag<any>,
-  LanguageButton: ComponentOrTag<any>,
-  UserRegistration: ComponentOrTag<any>,
-  Search: ComponentOrTag<any>,
-  SearchToggler: ComponentOrTag<any>,
-};
-type HeaderProps = DesignableComponentsProps<HeaderComponents> & HTMLProps<HTMLElement>;
+import { MenuTogglerClean } from '../MenuToggler';
+import { SearchTogglerClean, DesktopSearchClean } from '../Search';
+import { HeaderComponents, HeaderProps } from './types';
 
 const headerComponents: HeaderComponents = {
   Wrapper: Div,
   Container: Div,
   MenuContainer: Div,
-  MenuToggler: Fragment,
+  MenuToggler: MenuTogglerClean,
   Menu: Fragment,
   Logo: LogoClean,
-  UtilityMenu: Fragment,
-  SocialLinks: Fragment,
-  ButtonsMenuWrapper: Fragment,
-  ButtonsWrapper: Div,
-  Search: Fragment,
-  SearchToggler: Fragment,
+  UtilityMenu: Div,
+  DesktopSearch: DesktopSearchClean,
+  SearchToggler: SearchTogglerClean,
   LanguageButton: Fragment,
   UserRegistration: Fragment,
 };
 
-const HeaderCleanBase: FC<HeaderProps> = ({ components }) => {
-  const {
-    Wrapper,
-    Container,
-    MenuContainer,
-    MenuToggler,
-    Menu,
-    Logo,
-    UtilityMenu,
-    SocialLinks,
-    ButtonsMenuWrapper,
-    ButtonsWrapper,
-    LanguageButton,
-    UserRegistration,
-    Search,
-    SearchToggler,
-  } = components;
-
-  return (
-    <Wrapper>
-      <Container>
-        <MenuToggler />
-        <SearchToggler />
-        <Logo />
-        <UtilityMenu />
-        <SocialLinks />
-        <ButtonsMenuWrapper>
-          <ButtonsWrapper>
-            <Search />
-            <LanguageButton />
-            <UserRegistration />
-          </ButtonsWrapper>
-          <MenuContainer>
-            <Menu />
-          </MenuContainer>
-        </ButtonsMenuWrapper>
-      </Container>
-    </Wrapper>
-  );
-};
+const HeaderCleanBase: FC<HeaderProps> = ({ components: C }) => (
+  <C.Wrapper>
+    <C.Container>
+      <C.MenuToggler />
+      <C.Logo />
+      <C.SearchToggler />
+      <C.MenuContainer>
+        <C.Menu />
+        <C.UtilityMenu>
```

Comment:
hm... i'm not sure about this.  i think we will ultimately need to add arbitrarty links to the utility menu - so maybe this should be something like:
```
<UtilityMenuContainer>
  <C.DesktopSearch />
   <C.UtilityMenu />
    <C.LanguageSwitcher />
</UtiityMenuContainer>
``
the user registration is just a link which would go in the utility menu itself, while search and language switcher are probably unique components.

Reply:
You're right, I'll change this one.